builder/dockerfile/internals.go

```diff
@@ -383,11 +383,9 @@ func (b *Builder) calcCopyInfo(cmdName, origPath string, allowLocalDecompression
 		hfi.SetHash("file:" + hfi.Hash())
 		return copyInfos, nil
 	}
-
 	// Must be a dir
-
 	var subfiles []string
-	b.context.Walk(origPath, func(path string, info builder.FileInfo, err error) error {
+	b.context.Walk(statPath, func(path string, info builder.FileInfo, err error) error {
```

Comment:
Part of the problem with the bug is that the error was never checked here.  Should we at least warn if an error is encountered, if not fail completely?